File: gluecodium/src/test/resources/smoke/skip/output/android/com/example/smoke/SomeSkippedClass.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
import com.example.NativeBase;
import com.example.dont.smoke.DontSmokeEnum;
public final class SomeSkippedClass extends NativeBase {
    /**
     * For internal use only.
     * @hidden
     * @param nativeHandle The SDK nativeHandle instance.
     * @param dummy The SDK dummy instance.
     */
    protected SomeSkippedClass(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    @NonNull
    public native DontSmokeEnum doFoo();
}

File: gluecodium/src/test/resources/smoke/skip/output/android-kotlin/com/example/smoke/SomeSkippedClass.kt
/*

 *
 */

@file:JvmName("SomeSkippedClassExtensions")

package com.example.smoke

import com.example.NativeBase
import com.example.dont.smoke.DontSmokeEnum

class SomeSkippedClass : NativeBase {



    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}





    external fun doFoo() : DontSmokeEnum




    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}

